### PR TITLE
[TEP074] Remove CloudEvent pipelineResources

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit removes the CloudEvent Resources support.
Removal of `pipelineResources`, as in #5967 has been broken up into removal of each
resources packages for the `small PR` code standard.
This PR removes `github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1/cloudevent`
the respective example test and docs for cloud-event resources.

Part of #5967 
/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [n/a] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: please migrate off of `cloudevent` `pipelineresources` as it is removed, please refer to the doc at https://github.com/tektoncd/pipeline/blob/main/docs/pipelineresources.md#replacing-a-cloudevent-resource
```

### CODE REVIEW

File: third_party/github.com/cloudflare/circl/LICENSE
Comment (JeromeJu) [inline third_party/github.com/cloudflare/circl/LICENSE:1]: Running update-codegen on main to investigate these unexpected changes for the licenses.

File: pkg/reconciler/taskrun/taskrun_test.go
Comment (JeromeJu) [inline pkg/reconciler/taskrun/taskrun_test.go:3149]: I am under the impression that this whole test and CloudEvent could be removed along with the removal of CloudEvent resources.

File: pkg/reconciler/events/cloudevent/cloud_event_controller.go
Comment (lbernick) [inline pkg/reconciler/events/cloudevent/cloud_event_controller.go:41]: Are any other functions in this file now dead code?

File: pkg/reconciler/taskrun/taskrun_test.go
Comment (lbernick) [reply pkg/reconciler/taskrun/taskrun_test.go:3106]: yes, I think so

**Review by lbernick — APPROVED** [review]
@JeromeJu could you please add "action required" to the release notes, and a bit more detail on what's expected of users upgrading to the new release with these changes? (i.e. migrate off of cloudevent pipelineresources, with links to docs on how to do that)

File: pkg/reconciler/events/cloudevent/cloud_event_controller.go
Comment (JeromeJu) [reply pkg/reconciler/events/cloudevent/cloud_event_controller.go:41]: Thanks for the reminder, looking into it!

Comment (JeromeJu) [reply pkg/reconciler/events/cloudevent/cloud_event_controller.go:41]: Done, not any dead code that could be removed with CloudEvent resources.

Comment (afrittoli) [reply pkg/reconciler/events/cloudevent/cloud_event_controller.go:41]: There are two code paths for CloudEvents, one is associated with the pipeline resource and the other one with the CloudEvents feature.  If I remember correctly, `SendCloudEvents` is used by the former and `SendCloudEventsWithRetries` is used by the latter. 
I need to spend a bit more time reviewing the code to remember exactly what can be removed. I would recommend to try to remove the  `SendCloudEvents` and related usages and check if all cloud events test cases are still passing.

File: docs/resources.md
Comment (JeromeJu) [reply docs/resources.md]: Thanks for the suggestion. Done.

File: pkg/reconciler/events/cloudevent/cloud_event_controller.go
Comment (JeromeJu) [reply pkg/reconciler/events/cloudevent/cloud_event_controller.go]: Thanks Andrea, deleted:
- cloudEventDeliveryFromTargets
- SendCloudEvents

**Review by afrittoli — COMMENTED** [review]
Thank you @JeromeJu, this looks great. 

The 'cloudevent_count' metric counts cloudevent sent via the `PipelineResource`, so it doesn't make any more sense. I don't think we added metrics to the API stability policy, so it should be safe to remove. Besides, it wouldn't report anything at all.

I noticed that `taskRun.status.CloudEvents` has only just been marked as deprecated and is not planned for removal until Oct 2023. 

I though that when we deprecated `PipelineResources` we also deprecated all parts of the API associated with them, which would include resource inputs/outputs, resource results and cloudevents status. It may be that we were not explicit about the parts of the API affected, but I'm not sure we should keep around an API which is not used by anything.

@tektoncd/core-maintainers thoughts about this?

I don't think this discussion should block this PR, so it's fine to move ahead and possibly further cleanup in a follow-up PR. Perhaps here we could remove the metric at least?

**Comment by JeromeJu** [comment]
Many thanks to the detailed look into this removal and the respective effects on the related APIs. I have done the following 
so this PR might get in more safely for now. :)
- Tracked this down in the original issue: https://github.com/tektoncd/pipeline/pull/6002
- Put into the pipeline WG as a discussion

**Comment by JeromeJu** [comment]
Many thanks @afrittoli for the detailed look into this removal and the respective effects on the related APIs. I have done the following 
so this PR might get in more safely for now. :)
- Tracked this down in the original issue: https://github.com/tektoncd/pipeline/pull/6002
- Put into the pipeline WG as a discussion

**Comment by vdemeester** [comment]
> I though that when we deprecated `PipelineResources` we also deprecated all parts of the API associated with them, which would include resource inputs/outputs, resource results and cloudevents status. It may be that we were not explicit about the parts of the API affected, but I'm not sure we should keep around an API which is not used by anything.
> 
> @tektoncd/core-maintainers thoughts about this?

That was my thought as well.

**Review by vdemeester — APPROVED** [review]

**Pull request merged**
